bin/setup.sh

```diff
@@ -1,14 +1,22 @@
 # setup lsstsw environment
 #
-# source this file from your ~/.bashrc
+# source this file from your ~/.bashrc or ~/.zshrc
 #
 # relative to <lsstsw>/bin/
-LSSTSW=$(cd "$(dirname "$BASH_SOURCE")/.."; pwd)
-
-if [[ ! -f "$LSSTSW/eups/current/bin/setups.sh" ]]; then
-    echo "error: eups not found in $LSSTSW/eups/current" 1>&2
-    echo "  you may need to [re]run bin/deploy to [re]deploy EUPS." 1>&2
-    return
+if [[ -z $ZSH_NAME ]]; then
```

Comment:
in this if block just define $LSSTSW and the file extension of the EUPS `setups.[z]sh` file. Then just use  `$SUFFIX` in later code. That will minimize all the duplication. Ironically, EUPS generates identical sh and zsh setup files.